components/product/ProductApplication/ProductApplications.tsx

```diff
@@ -0,0 +1,207 @@
+    setShowView(hasRelevantData)
+  }, [product])
+
+  // Generate updated array when dependencies change
+  useEffect(() => {
+    const updateNewArray = () => {
+      if (!applicationDilutionRange.length || !themeCodeMapping.length) return
+
+      const updatedArray = applicationDilutionRange.map((data) => {
+        const matchedItem = themeCodeMapping.find(
+          (item) => data?.Application?.toLowerCase() === item?.applId?.toLowerCase()
+        )
+        return matchedItem
+          ? {
+              Application: matchedItem?.longName,
+              ApplicationDilutionRange: data?.ApplicationDilutionRange,
+            }
+          : data
+      })
+
+      setNewArray(updatedArray)
+    }
+
+    updateNewArray()
+  }, [applicationDilutionRange, themeCodeMapping])
+
+  // Fetch theme settings on mount
+  useEffect(() => {
+    async function fetchThemeSettings() {
+      const settings = await GetThemeSettings()
+      setThemeCodeMapping(settings?.data?.themeCodeMapping || [])
+    }
+    fetchThemeSettings()
+  }, [])
+
+  return (
+    <Box sx={{ width: '100%', display: 'flex', justifyContent: 'center' }}>
+      {showView && (
+        <Box
+          sx={{
+            width: '100%',
+            maxWidth: '1200px',
+          }}
+        >
+          <Typography variant="h3" sx={{ marginBottom: '10px' }}>
+            Applications
+          </Typography>
+
+          {applicationText && (
+            <Box
+              sx={{
+                marginBottom: '22px',
+                color: `${grey[900]}`,
+              }}
+            >
+              <Typography
+                variant="body2"
+                gutterBottom
+                sx={{ color: '#020027' }}
+                dangerouslySetInnerHTML={{
+                  __html: applicationText,
+                }}
+              />
+            </Box>
+          )}
+
+          {applicationTextVariant && (
+            <Box
+              sx={{
+                marginBottom: '22px',
+                color: `${grey[900]}`,
+              }}
+            >
+              <Typography
+                variant="body2"
+                gutterBottom
+                sx={{ color: '#020027' }}
+                dangerouslySetInnerHTML={{
+                  __html: applicationTextVariant,
+                }}
+              />
+            </Box>
+          )}
+
+          {applicationDilutionRange && (
+            <TableContainer
+              component={Paper}
+              sx={{
+                borderRadius: 0,
+                boxShadow: '0px 1px 0px rgba(0,0,0,0.14),0px 1px 0px rgba(0,0,0,0.12)',
+              }}
+            >
+              <Table>
+                <TableBody>
+                  {newArray.map((data: any, index: number) => (
+                    <TableRow key={index}>
+                      <TableCell
+                        variant="head"
+                        sx={{
+                          width: '25%',
```

Comment:
do not use inline styling, or combine the common properties into object and then use it.